src/java/org/apache/cassandra/service/StartupChecks.java

```diff
@@ -285,6 +286,49 @@ public void execute()
         }
     };
 
+    public static final StartupCheck checkReadAheadKbSetting = new StartupCheck()
+    {
+        // This value is in KB.
+        private static final long EXPECTED_READ_AHEAD_KB_SETTING = 8;
+        private static final String READ_AHEAD_KB_SETTING_PATH = "/sys/block/sda/queue/read_ahead_kb";
+
+        @Override
+        public void execute() throws StartupException
+        {
+            if (!FBUtilities.isLinux)
+                return;
+
+            final Path path = Paths.get(READ_AHEAD_KB_SETTING_PATH);
+            try (final BufferedReader bufferedReader = Files.newBufferedReader(path))
+            {
+                final String data = bufferedReader.readLine();
+                if (data != null)
+                {
+                    try
+                    {
+                        Integer readAheadKbSetting = Integer.parseInt(data);
+
+                        if (readAheadKbSetting > EXPECTED_READ_AHEAD_KB_SETTING)
+                        {
+                            logger.warn("High Read Ahead Kb setting: It is Recommended to set this value to 8KB " +
```

Comment:
Maybe prefix the message with: `Detected high 'read_ahead_kb' setting for device  'sda' of data directory '/srv/disk01'.  It is Recommended to set this value to 8KB....`

Reply:
Changed, thanks